batch.go

```diff
@@ -410,7 +423,7 @@ func (batch *Batch) build() error {
 		bcADV.TotalCreditEntryDollarAmount, bcADV.TotalDebitEntryDollarAmount = batch.calculateADVBatchAmounts()
 		batch.ADVControl = bcADV
 	}
-	return nil
+	return batch.upsertOffset()
```

Comment:
upsert means insert right?

Reply:
Nope. `upsert` is to "update or insert". It's an idempotent operation on the Batch that always leaves an offset record as the last EntryDetail.

Reply:
Thanks for the clarification